### PR TITLE
Feature/sof 6814

### PR DESCRIPTION
**Pull request by VsevolodX**
Made elements labels to be 3D aware and occluded by atoms, removed useless code.

### CODE REVIEW

File: src/mixins/labels.js [hunk 1]
```diff
@@ -71,51 +71,40 @@ export const LabelsMixin = (superclass) =>
          */
         createLabelSprite(text, name) {
             const texture = this.getLabelTextTexture(text);
-            const spriteMaterial = new THREE.SpriteMaterial({ map: texture });
+            const spriteMaterial = new THREE.SpriteMaterial({
+                map: texture,
+                transparent: true,
+                depthFunc: THREE.LessEqualDepth,
+                depthTest: true,
```
Comment (timurbazhirov) [inline src/mixins/labels.js]: Move to settings

File: src/settings.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ export default {
     labelPointsConfig: {
         size: 1.5,
         depthTest: true,
-        depthFunc: THREE.NotEqualDepth,
+        depthFunc: THREE.LessEqualDepth,
```
Comment (timurbazhirov) [inline src/settings.js]: Do we need to modify this??

File: src/wave.js [hunk 3]
```diff
@@ -322,12 +323,13 @@ export class Wave extends mix(WaveBase).with(
         this.drawBoundaries();
         if (this.isDrawBondsEnabled) this.drawBonds();
         this.render();
-        this.createLabelSpritesAsPoints();
+        if (this.fastLabelsRender) this.createLabelsAsPoints();
```
Comment (timurbazhirov) [inline src/wave.js]: Let's have this logic instead a function `createLabels` and abstract it from here

File: src/mixins/labels.js [hunk 4]
```diff
@@ -173,27 +159,46 @@ export const LabelsMixin = (superclass) =>
...
+                    const atomPosition = new THREE.Vector3().fromArray(vertices, i);
+                    const labelSprite = this.createLabelSprite(key, `label-for-${key}`);
+                    const offsetVector = this.getLabelOffsetVector(atomPosition, key);
+                    labelSprite.userData = { atomPosition, atomName: key };
+                    labelSprite.position.addVectors(atomPosition, offsetVector);
+                    this.labelsGroup.add(labelSprite);
+                }
+            });
+            this.structureGroup.add(this.labelsGroup);
+            this.render();
```
Comment (timurbazhirov) [inline src/mixins/labels.js]: Per my comment below `this.render()` should go to `createLabels` function

File: src/mixins/labels.js (hunk 1 above)
Comment (VsevolodX) [reply src/mixins/labels.js]: OK

File: src/settings.js (hunk 2 above)
Comment (VsevolodX) [reply src/settings.js]: For points it should be as it was before, I'll revert

File: src/wave.js [hunk 5]
```diff
@@ -322,12 +322,12 @@ export class Wave extends mix(WaveBase).with(
...
         if (this.isDrawBondsEnabled) this.drawBonds();
         this.render();
-        this.createLabelSpritesAsPoints();
+        this.createLabels();
         this.refillChosenAtoms();
     }
 
     render() {
-        this.adjustLabelsToCameraPosition(this.scene, this.camera);
+        if (this.settings.labelsConfig.areSpritesUsed) this.adjustLabelsToCameraPosition();
```
Comment (timurbazhirov) [inline src/wave.js]: Move this to create labels?

Comment (VsevolodX) [reply src/wave.js]: No. Looked at it again. Adjustment should happen on render() after the camera moved, not when the labels are created.

Comment (timurbazhirov) [reply src/wave.js]: Let's move the if statement to the function itself

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**